contracts/governance/TimelockControllerEnumerable.sol

```diff
@@ -0,0 +1,198 @@
+    function scheduleBatch(
+        address[] calldata targets,
+        uint256[] calldata values,
+        bytes[] calldata payloads,
+        bytes32 predecessor,
+        bytes32 salt,
+        uint256 delay
+    ) public virtual override {
+        super.scheduleBatch(targets, values, payloads, predecessor, salt, delay);
+        bytes32 id = hashOperationBatch(targets, values, payloads, predecessor, salt);
+        _operationsBatchIdSet.add(id);
+        _operationsBatchMap[id] = OperationBatch({
+            targets: targets,
+            values: values,
+            payloads: payloads,
+            predecessor: predecessor,
+            salt: salt,
+            delay: delay
+        });
+    }
+
+    /// @inheritdoc TimelockController
+    function cancel(bytes32 id) public virtual override {
+        super.cancel(id);
+        if (_operationsIdSet.contains(id)) {
+            _operationsIdSet.remove(id);
+            delete _operationsMap[id];
+        }
+        if (_operationsBatchIdSet.contains(id)) {
+            _operationsBatchIdSet.remove(id);
+            delete _operationsBatchMap[id];
+        }
+    }
+
+    /// @notice Return all scheduled operations
+    /// @return operations_ The operations array
+    function operations() public view returns (Operation[] memory operations_) {
+        uint256 operationsCount_ = _operationsIdSet.length();
+        operations_ = new Operation[](operationsCount_);
+        for (uint256 i = 0; i < operationsCount_; i++) {
+            operations_[i] = _operationsMap[_operationsIdSet.at(i)];
+        }
+        return operations_;
+    }
+
+    /// @notice Return the number of operations from the set
+    /// @return operationsCount_ The number of operations
+    function operationsCount() public view returns (uint256 operationsCount_) {
+        operationsCount_ = _operationsIdSet.length();
+        return operationsCount_;
+    }
+
+    /// @notice Return the operation at the given index
+    /// @param index The index of the operation
+    /// @return operation_ The operation
+    function operation(uint256 index) public view returns (Operation memory operation_) {
+        if (index >= _operationsIdSet.length()) {
+            revert OperationIndexNotFound(index);
+        }
+        operation_ = _operationsMap[_operationsIdSet.at(index)];
+        return operation_;
+    }
+
+    /// @notice Return the operation with the given id
+    /// @param id The id of the operation
+    /// @return operation_ The operation
+    function operation(bytes32 id) public view returns (Operation memory operation_) {
+        if (!_operationsIdSet.contains(id)) {
+            revert OperationIdNotFound(id);
+        }
+        operation_ = _operationsMap[id];
+        return operation_;
+    }
+
+    /// @notice Return all scheduled operation batches
+    /// @return operationsBatch_ The operationsBatch array
+    function operationsBatch() public view returns (OperationBatch[] memory operationsBatch_) {
```

Comment:
I would add the same warning as suggested for `operations()`. Also same suggestion about receiving a range as a arguments